scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/DBJobDataParameters.java

```diff
@@ -85,17 +85,17 @@ public class DBJobDataParameters {
         this.parentId = parentId;
         this.sortParameters = sortParameters;
 
-        Set<JobStatus> newStatus = new HashSet<JobStatus>();
+        Set<Integer> newStatusRanks = new HashSet<Integer>();
         if (pending) {
-            newStatus.addAll(SchedulerDBManager.PENDING_JOB_STATUSES);
+            newStatusRanks.add(JobState.PENDING_RANK);
         }
         if (running) {
-            newStatus.addAll(SchedulerDBManager.RUNNING_JOB_STATUSES);
+            newStatusRanks.add(JobState.RUNNING_RANK);
         }
         if (finished) {
-            newStatus.addAll(SchedulerDBManager.FINISHED_JOB_STATUSES);
+            newStatusRanks.add(JobState.FINISHED_RANK);
         }
```

Comment:
Is it possible to use `else if ` to avoid executing additional if statements?

Reply:
no because it is not exclusive conditions. you can have all possible combinations of pending, running and finished